### PR TITLE
Consolidate get_shipping_providers to always return AbstractShippingProvider instances

### PR DESCRIPTION
**Pull request by tpaksu**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Consolidates `FulfillmentUtils::get_shipping_providers()` and `FulfillmentUtils::get_shipping_providers_object()` into a single method that always returns `AbstractShippingProvider[]` instances keyed by provider key.

**Problem:** `get_shipping_providers()` returned a mixed array of class name strings and `AbstractShippingProvider` instances. Every caller had to duplicate the same `is_string` check + DI container instantiation logic before using the providers. `get_shipping_providers_object()` existed solely to do this conversion for UI consumers.

**Changes:**
- `get_shipping_providers()` now resolves class name strings from the filter into `AbstractShippingProvider` instances internally (backward-compatible with filters that still return strings)
- Removed `get_shipping_providers_object()` entirely
- Simplified `FulfillmentsManager::try_parse_tracking_number()` by removing redundant `is_string` checks and instantiation logic (reduced from 25 lines to 5)
- Simplified `class-wc-ajax.php` built-in provider key collection by using `instanceof CustomShippingProvider` instead of string class checks
- Updated `FulfillmentsRenderer` (4 call sites) and REST `Controller::get_providers()` to build UI arrays from provider instances directly
- Updated tests to assert on `AbstractShippingProvider` instances instead of class name strings

Closes https://linear.app/a8c/issue/WOOPLUG-6533 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://developer.woocommerce.com/docs/contribution/testing/writing-high-quality-testing-instructions/), include your detailed testing instructions:

1. Enable the fulfillments feature flag: `wp option update woocommerce_feature_fulfillments_enabled yes`
2. Go to **WooCommerce > Orders** and verify the shipping provider filter dropdown still displays all providers correctly.
3. Create/edit an order, add a fulfillment with a tracking number. Verify that auto-detection of the shipping provider still works (e.g., enter a UPS or FedEx tracking number).
4. Verify the fulfillment drawer shows the correct provider name, icon, and tracking URL.
5. Go to **WooCommerce > Settings > Fulfillments > Shipping Providers** and add a custom shipping provider. Verify it appears in the provider dropdown and is correctly excluded from the "built-in" provider list.
6. Use the REST API endpoint `GET /wc/v4/fulfillments/providers` and verify it returns the expected provider data structure (`label`, `icon`, `value`, `url`).

### Testing that has already taken place:

- PHPUnit tests: `ShippingProvidersTest` (1 test, 231 assertions), `FulfillmentsManagerTest` (17 tests, 42 assertions), `ProvidersTest` (194 tests, 863 assertions) all passing.
- PHP lint (`phpcs-changed`): clean.
- PHPStan: no errors on all 5 modified source files.

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [x] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Consolidate get_shipping_providers and get_shipping_providers_object into a single method that always returns AbstractShippingProvider instances.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/RestApi/Routes/V4/Fulfillments/Controller.php [hunk 1]
```diff
@@ -436,7 +436,15 @@ private function prepare_error_response( $code, $message, $data ): WP_REST_Respo
...
 	public function get_providers( WP_REST_Request $request ): WP_REST_Response {
-		$providers = \Automattic\WooCommerce\Admin\Features\Fulfillments\FulfillmentUtils::get_shipping_providers_object();
+		$providers = array();
+		foreach ( \Automattic\WooCommerce\Admin\Features\Fulfillments\FulfillmentUtils::get_shipping_providers() as $provider ) {
+			$providers[ $provider->get_key() ] = array(
+				'label' => $provider->get_name(),
+				'icon'  => $provider->get_icon(),
+				'value' => $provider->get_key(),
+				'url'   => $provider->get_tracking_url( '__PLACEHOLDER__' ),
+			);
```
[the remark this answers is not in the document]
Comment (tpaksu) [reply plugins/woocommerce/src/Internal/RestApi/Routes/V4/Fulfillments/Controller.php:446]: Fixed in 2b6c4523de. Added `?? ''` to both `Controller.php` and `FulfillmentsRenderer.php`. Also added `assertNotEmpty` to the test and removed the stale PHPStan baseline entry.

**Review by ayushpahwa — APPROVED** [review]
Changes look good and the testing instructions work as expected! Thanks for removing the dupes, LGTM 🚀
